src/Bicep.Cli/Commands/BuildParamsCommand.cs

```diff
@@ -58,49 +58,45 @@ public async Task<int> RunAsync(BuildParamsArguments args)
                 logger.LogWarning(CliResources.ResourceTypesDisclaimerMessage);
             }
 
-            if(bicepInputPath != null && !IsBicepFile(bicepInputPath))
+            if(bicepFileArgPath != null && !IsBicepFile(bicepFileArgPath))
             {
-                throw new CommandLineException($"{bicepInputPath} is not a bicep file");
+                throw new CommandLineException($"{bicepFileArgPath} is not a bicep file");
             }
 
             if(features.ParamsFilesEnabled && IsBicepparamsFile(paramsInputPath)) 
```

Comment:
I'd recommend doing the validation before this condition check - that way you don't have to duplicate it and can reduce nesting:

```csharp
if (!features.ParamsFilesEnabled) 
{
    logger.LogError(CliResources.UnableToCompileParamsFile, paramsInputPath, nameof(ExperimentalFeaturesEnabled.ParamsFiles));
    return 1;
}
```
Or by throwing:
```csharp
if (!features.ParamsFilesEnabled) 
{
    throw new CommandLineException(
        string.Format(CliResources.UnableToCompileParamsFile, paramsInputPath, nameof(ExperimentalFeaturesEnabled.ParamsFiles)));
}
```